xml/hwbp_registers.xml

```diff
@@ -289,4 +292,85 @@
             Other values are reserved for future use.
         </field>
     </register>
+
+    <register name="Exception Trigger" short="etrigger" address="0x7a1">
+        This register is accessible as \Rtdataone when \Ftype is 4.
+
+        If implemented, it allows triggering on a variety of interrupt and
+        exception sources. If such a trigger is taken, all the usual CSRs as
+        well as \Rpc are updated, and the requested action is taken just before
+        the first instruction of the interrupt/exception handler is executed.
+
+        <field name="type" bits="XLEN-1:XLEN-4" access="R" reset="2" />
+        <field name="dmode" bits="XLEN-5" access="R/W" reset="0" />
+        <field name="0" bits="XLEN-6:24" access="R" reset="0" />
+        <field name="source" bits="23:10" access="R/W" reset="0">
+            For each bit that is set, the corresponding interrupt/exception
+            will cause this trigger to fire if bit corresponding to the current
+            mode is also set.
+
+            0: Software interrupt
+
+            1: Timer interrupt
+
+            2: External interrupt
+
+            3: Instruction address misaligned
+
+            4: Instruction access fault
+
+            5: Illegal instruction
+
+            6: Load address misaligned
+
+            7: Load access fault
+
+            8: Store/AMO address misaligned
+
+            9: Store/AMO access fault
+
+            10: Environment call
+
+            11: Instruction page fault
+
+            12: Load page fault
+
+            13: Store/AMO page fault
+        </field>
+        <field name="m" bits="9" access="R/W" reset="0">
+            This trigger only fires in M mode if this bit is 1.
```

Comment:
why the "this trigger only fires in only" vs "this trigger fires in"? What if all 3 bits are set, the logic makes no sense

Reply:
Sounds like "only" is a poor word choice there. The intent is that if you want the trigger to fire in M mode, this bit has to be 1. Otherwise it won't. If you also want it to fire in S mode, you also need to set the S bit.

Reply:
Oh. Yeah I read that wrong. How about "This trigger fires in M mode if and only if this bit is 1."

Reply:
Actually, let me just copy from \Rmcontrol:
When set, enable this trigger in S mode.